packages/interbit-ui-tools/src/tests/middleware/sagas.test.js

```diff
@@ -0,0 +1,588 @@
+      .put.like({
+        action: { type: actionTypes.INTERBIT_READY }
+      })
+      .put.like({
+        action: { type: actionTypes.STATIC_CHAINS_SAGA }
+      })
+      .not.put.like({
+        action: { type: actionTypes.INTERBIT_LOADING }
+      })
+      .not.put.like({
+        action: {
+          type: actionTypes.INTERBIT_ERROR
+        }
+      })
+      .not.put.like({
+        action: {
+          type: actionTypes.CHAIN_ERROR
+        }
+      })
+      .run()
+  })
+
+  it('If interbit is not loaded, connect to interbit and dispatch STATIC_CHAINS_SAGA action', async () => {
+    const unconnectedContext = mockContext({
+      isInterbitLoaded: false
+    })
+    const { rootSaga } = middlewareSagas(unconnectedContext)
+    await expectSaga(rootSaga)
+      .withReducer(reducers)
+      .put.like({
+        action: {
+          type: actionTypes.INITIAL_CONFIG,
+          payload: configWithPublicChain
+        }
+      })
+      .put.like({
+        action: { type: actionTypes.INTERBIT_LOADING }
+      })
+      .put.like({
+        action: {
+          type: actionTypes.INTERBIT_PUBLIC_KEY,
+          payload: { publicKey: PUBLIC_KEY }
+        }
+      })
+      .put.like({
+        action: {
+          type: actionTypes.INTERBIT_LOADED,
+          payload: { version: VERSION }
+        }
+      })
+      .put.like({
+        action: { type: actionTypes.INTERBIT_READY }
+      })
+      .put.like({
+        action: { type: actionTypes.STATIC_CHAINS_SAGA }
+      })
+      .not.put.like({
+        action: {
+          type: actionTypes.INTERBIT_ERROR
+        }
+      })
+      .not.put.like({
+        action: {
+          type: actionTypes.CHAIN_ERROR
+        }
+      })
+      .run()
+  })
+
+  it('If loading interbit throws, dispatch INTERBIT_ERROR action', async () => {
+    const errorMessage = 'interbit is unavailable today'
+    const failingContext = {
+      ...mockContext({ isInterbitLoaded: false }),
+      waitForInterbit: async () => {
+        throw new Error(errorMessage)
+      }
+    }
+    const { rootSaga } = middlewareSagas(failingContext)
+    await expectSaga(rootSaga)
+      .withReducer(reducers)
+      .put.like({
+        action: { type: actionTypes.INTERBIT_LOADING }
+      })
+      .put.like({
+        action: {
+          type: actionTypes.INTERBIT_ERROR,
+          payload: { error: errorMessage }
+        }
+      })
+      .not.put.like({
+        action: { type: actionTypes.STATIC_CHAINS_SAGA }
+      })
+      .run()
+  })
+
+  it('If no static chains are configured, no problem', async () => {
+    const contextWithoutStaticChains = mockContext({
+      config: configWithoutStaticChains
+    })
+    const { rootSaga } = middlewareSagas(contextWithoutStaticChains)
+    await expectSaga(rootSaga)
+      .withReducer(reducers)
+      .put.like({
+        action: {
+          type: actionTypes.INITIAL_CONFIG,
+          payload: configWithoutStaticChains
+        }
+      })
+      .put.like({
```

Comment:
Would it be reasonable to also specify `.not.put.like({...CHAIN_LOADING})` here or will this attempt to load a private chain?

Reply:
`.not.put.like()` is an assertion about the behaviour and has no side effects. So this makes sense.

Reply:
Done